src/Stormpath/Resource/Resource.php

```diff
@@ -87,6 +87,21 @@ public function getPropertyNames()
         return array_keys((array) $this->properties);
     }
 
+    public function getDirtyPropertyNames()
```

Comment:
Instead of creating a new method and having to expose other methods, etc., I would just add an optional boolean argument to the `getPropertyNames()` method. I would call it `$retrieveDirtyProperties` and implement the logic from there. Just evaluate whether we should default this value to be `true` or `false`, according to what's use the most on the SDK.